tests/unit/package_managers/test_gomod.py

```diff
@@ -509,18 +582,274 @@ def test_parse_broken_go_sum(rooted_tmp_path: RootedPath, caplog: pytest.LogCapt
 
     submodule = rooted_tmp_path.join_within_root("submodule")
     submodule.path.mkdir()
-    submodule.join_within_root("go.sum").path.write_text(go_sum_content)
+    go_sum_file = submodule.join_within_root("go.sum")
+    go_sum_file.path.write_text(go_sum_content)
 
-    assert _parse_go_sum(submodule) == expect_modules
+    assert _parse_go_sum(go_sum_file) == expect_modules
     assert caplog.messages == [
         "submodule/go.sum:2: malformed line, skipping the rest of the file: 'github.com/davecgh/go-spew v1.1.0/go.mod'",
     ]
 
 
+@mock.patch("cachi2.core.package_managers.gomod.Go")
 @mock.patch("cachi2.core.package_managers.gomod.ModuleVersionResolver")
-def test_create_modules_from_parsed_data(mock_version_resolver: mock.Mock, tmp_path: Path) -> None:
+def test_parse_local_modules(go: mock.Mock, version_resolver: mock.Mock) -> None:
+    go.return_value = """
+    {
+        "Path": "myorg.com/my-project",
+        "Main": true,
+        "Dir": "/path/to/project"
+    }
+    {
+        "Path": "myorg.com/my-project/workspace",
+        "Main": true,
+        "Dir": "/path/to/project/workspace"
+    }
+    """
+
+    app_dir = RootedPath("/path/to/project")
+    version_resolver.get_golang_version = lambda _, __: "1.0.0"
```

Comment:
Any better way to mock the result of this function?

Reply:
```
diff --git a/tests/unit/package_managers/test_gomod.py b/tests/unit/package_managers/test_gomod.py
index 5f617d33..7362f851 100644
--- a/tests/unit/package_managers/test_gomod.py
+++ b/tests/unit/package_managers/test_gomod.py
@@ -608,7 +608,7 @@ def test_parse_local_modules(go: mock.Mock, version_resolver: mock.Mock) -> None
     """
 
     app_dir = RootedPath("/path/to/project")
-    version_resolver.get_golang_version = lambda _, __: "1.0.0"
+    version_resolver.get_golang_version.return_value = "1.0.0"
 
     main_module, workspace_modules = _parse_local_modules(go, [], {}, app_dir, version_resolver)
```